jax/_src/config.py

```diff
@@ -1449,6 +1449,57 @@ def _update_disable_jit_thread_local(val):
     update_thread_local_hook=lambda val: \
       update_thread_local_jit_state(default_matmul_precision=val))
 
+_logging_level_handler_set: dict[str, tuple[logging.Handler, int]] = {}
+
+def _update_logging_level_global(logging_level: str | None):
+  # remove previous handlers
+  for logger_name, (handler, level) in _logging_level_handler_set.items():
+    logger = logging.getLogger(logger_name)
+    logger.removeHandler(handler)
+    logger.setLevel(level)
+  _logging_level_handler_set.clear()
+
+  if logging_level is None:
+    return
+
+  # attempt to convert the logging level to integer
+  try:
+    # logging level is a string representation of an integer
+    logging_level_num = int(logging_level)
+  except ValueError:
+    # logging level is a name string
+    logging_level_num = logging.getLevelName(logging_level)
+
+  # configure the CPP logging level 0 - debug, 1 - info, 2 - warning, 3 - error
+  cpp_log_level = min(max(0, (logging_level_num // 10) - 1), 3)
+  os.environ["TF_CPP_MIN_LOG_LEVEL"] = str(cpp_log_level)
+
+  handler = logging.StreamHandler()
+  handler.setLevel(logging_level_num)
+  handler.setFormatter(logging_config.logging_formatter)
+
+  # update jax and jaxlib root loggers for propagation
+  root_loggers = [logging.getLogger("jax"), logging.getLogger("jaxlib")]
+  for logger in root_loggers:
+    logger.setLevel(logging_level_num)
+    logger.addHandler(handler)
+    _logging_level_handler_set[logger.name] = (handler, logger.level)
+
+
+# Don't define a context manager since this isn't threadsafe.
+optional_enum_state(
+    name='jax_logging_level',
+    enum_values=['NOTSET', 'DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL',
+                 '0', '10', '20', '30', '40', '50'],
```

Comment:
Maybe we should just accept the strings? The numbers are kind of confusing since you can't choose anything in between. I'm fine with this as-is though, I find Python logging bizarre...

Reply:
Yes, fair, let's just do strings!